core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,67 +351,3183 @@ def validate(cls, item):
+            cls.errors['contributors summary check'].append((
+                'Entity id %s: Contributor ids: %s do not match the '
+                'contributor ids obtained using contributors summary: %s') % (
+                    item.id, (',').join(sorted(item.contributor_ids)),
+                    (',').join(
+                        sorted(contributor_ids_from_contributors_summary))))
+
+    @classmethod
+    def _validate_language_code(cls, item):
+        """Validate that language code is present in allowed language codes.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        if not utils.is_valid_language_code(item.language_code):
+            cls.errors['language code check'].append((
+                'Entity id %s: Language code %s for model is unsupported' % (
+                    item.id, item.language_code)))
+
+    @classmethod
+    def _validate_node_count(cls, item):
+        """Validate that node_count of model is equal to number of nodes
+        collection_contents in collection model.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        _, collection_model_tuples_list = cls.external_models[
+            'collection_ids']
+
+        for (_, collection_model) in collection_model_tuples_list:
+            nodes = collection_model.collection_contents['nodes']
+            if item.node_count != len(nodes):
+                cls.errors['node count check'].append((
+                    'Entity id %s: Node count: %s does not match the number of '
+                    'nodes in collection_contents dict: %s') % (
+                        item.id, item.node_count, len(nodes)))
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that model properties match the corresponding collection
+        model and collection rights model properties.
+
+        Args:
+            item: CollectionSummaryModel to validate.
+        """
+        _, collection_model_tuples_list = cls.external_models[
+            'collection_ids']
+        _, collection_rights_model_tuples_list = cls.external_models[
+            'collection_rights_ids']
+
+        collection_model_properties_dict = {
+            'title': 'title',
+            'category': 'category',
+            'objective': 'objective',
+            'language_code': 'language_code',
+            'tags': 'tags',
+            'collection_model_created_on': 'created_on',
+        }
+
+        for (_, collection_model) in collection_model_tuples_list:
+            for (property_name, collection_model_property_name) in (
+                    collection_model_properties_dict.iteritems()):
+                value_in_summary_model = getattr(item, property_name)
+                value_in_collection_model = getattr(
+                    collection_model, collection_model_property_name)
+
+                summary_model_output_value = value_in_summary_model
```

Comment:
The name of this and the next variable should have the concept of "as_str" somewhere in it. Also you should probably define it under the "if" in line 1035, otherwise it's unnecessary to compute these.

I am not actually certain you need to even do this though. What happens if you interpolate a list in a string? Does it auto-cast correctly? I just tried and it seems to...

Ditto elsewhere that you do something like this.

Reply:
It does auto-cast correctly but the output looks a bit messy, so I thought it would be better to have a comma-separated list of values as a string here. What do you suggest?

Reply:
This is just for internal admin use, so I think it's OK to do the simpler thing if it's readable by the developer who needs to fix any issues -- i.e. you :P That's because it results in less code to maintain. (In the worst case, you could run a script over the output to make things pretty.) Ultimately, it's your decision, since you'll be using the output.

On the other hand, try to ensure that there's no data loss (so that we don't need to modify stuff and run the job again, which takes time).

Reply:
Yes, I agree with this, it is better to use list directly here since anyways I am going to use the output :P
Updated now.